### PR TITLE
Update Release notes for  odatalib-7.7.1

### PR DESCRIPTION
**Pull request by Sreejithpin**
Release notes for  odatalib-77.1

### CODE REVIEW

File: Odata-docs/changelog/odatalib-7x.md [hunk 1]
```diff
@@ -29,6 +29,44 @@ There was a lot of vestigial code left around from the OData v1-3 days that we
...
+
+[[#1793](https://github.com/OData/odata.net/pull/1793)] Different null validation messages for complex and primitive collections
+
+***Improvements***
+
+[[#1853](https://github.com/OData/odata.net/pull/1853)] Removed the unnecessary string allocations from hot path in JsonReader
+
+[[#1849](https://github.com/OData/odata.net/pull/1849)] Add "RemovalDate" property to deprecation annotation
+
+[[#1840](https://github.com/OData/odata.net/pull/1840)] update client E2E xunit tests
```
Comment (xuzhg) [inline Odata-docs/changelog/odatalib-7x.md:60]: >u [](start = 56, length = 1)

upper "U"

File: Odata-docs/changelog/odatalib-7x.md [hunk 2]
```diff
@@ -29,6 +29,44 @@ There was a lot of vestigial code left around from the OData v1-3 days that we
...
+
+[[#1831](https://github.com/OData/odata.net/pull/1831)] Fix bug on atomicityGroup as a dependsOn value in Odata v4
+
+[[#1855](https://github.com/OData/odata.net/pull/1855)] Hotfix for the infinite loop if structured type has property which type is the declaring type
+
+[[#1865](https://github.com/OData/odata.net/pull/1865)] Fix Enum as string used with Has Issue
+
+[[#1762](https://github.com/OData/odata.net/pull/1762)] Enable Where clause to generate $filter query options for key predicates
+
+[[#1793](https://github.com/OData/odata.net/pull/1793)] Different null validation messages for complex and primitive collections
```
Comment (xuzhg) [inline Odata-docs/changelog/odatalib-7x.md:52]: sort the PR, Issue number.

File: Odata-docs/changelog/odatalib-7x.md (hunk 1 above)
Comment (Sreejithpin) [reply Odata-docs/changelog/odatalib-7x.md]: Done

File: Odata-docs/changelog/odatalib-7x.md (hunk 2 above)
Comment (Sreejithpin) [reply Odata-docs/changelog/odatalib-7x.md]: Done

**Pull request merged**